relay/src/main.rs

```diff
@@ -32,25 +39,35 @@ pub enum Command {
         /// The input to provide to the guest binary
         input: Option<String>,
     },
-    Relay {
-        /// Ethereum Proxy address
-        #[arg(short, long)]
-        relay_contract_address: Address,
-
+    /// Upload the RISC-V ELF binary to Bonsai.
+    Upload {
+        /// The name of the guest binary
+        guest_binary: String,
+        /// Bonsai API URL
+        bonsai_api_url: String,
+        /// Bonsai API URL
+        bonsai_api_key: String,
+    },
+    /// Upload the RISC-V ELF binary to Bonsai.
+    Run {
+        /// Bonsai API URL
+        #[arg(long, env)]
+        bonsai_api_url: String,
+        /// Bonsai API URL
+        #[arg(long, env)]
+        bonsai_api_key: String,
+        /// Bonsai Relay contract address on Ethereum
+        #[arg(long, env)]
+        relay_address: Address,
         /// Ethereum Node endpoint
-        #[arg(long)]
-        eth_node_url: String,
-
-        /// Ethereum Chain ID
-        #[arg(long, default_value_t = 31337)]
+        #[arg(long, env)]
+        eth_node: String,
+        /// Ethereum chain ID
+        #[arg(long, default_value_t = 5)]
```

Comment:
I would suggest that the default should be `None` and to have it automatically looked up from the remote node if not found. Otherwise this may add some friction to local deployment.